java/google/registry/rde/XjcToDomainResourceConverter.java

```diff
@@ -0,0 +1,329 @@
+        .setCreationClientId(domain.getCrRr().getValue())
+        .setCreationTime(domain.getCrDate())
+        .setRegistrationExpirationTime(domain.getExDate())
+        .setLastEppUpdateTime(domain.getUpDate())
+        .setLastEppUpdateClientId(domain.getUpRr() == null ? null : domain.getUpRr().getValue())
+        .setLastTransferTime(domain.getTrDate())
+        .setStatusValues(
+            ImmutableSet.copyOf(Iterables.transform(domain.getStatuses(), STATUS_CONVERTER)))
+        .setNameservers(convertNameservers(domain.getNs()))
+        .setGracePeriods(ImmutableSet
+            .copyOf(Iterables.transform(
+                domain.getRgpStatuses(), gracePeriodConverter)))
+        .setContacts(
+            ImmutableSet.copyOf(Iterables.transform(domain.getContacts(), CONTACT_CONVERTER)))
+        .setDsData(
+            domain.getSecDNS() == null ? ImmutableSet.<DelegationSignerData>of() : ImmutableSet
+            .copyOf(Iterables.transform(domain.getSecDNS().getDsDatas(), SECDNS_CONVERTER)))
+        .setTransferData(convertDomainTransferData(domain.getTrnData()));
+    checkArgument(
+        domain.getRegistrant() != null, "Registrant is missing for domain '%s'", domain.getName());
+    builder = builder.setRegistrant(convertRegistrant(domain.getRegistrant()));
+    return builder.build();
+  }
+
+  private static BillingEvent.Recurring createAutoRenewBillingEvent(
+    XjcRdeDomain domain, HistoryEntry historyEntry) {
+    final BillingEvent.Recurring billingEvent = new BillingEvent.Recurring.Builder()
+        .setReason(Reason.RENEW)
+        .setFlags(ImmutableSet.of(Flag.AUTO_RENEW))
+        .setTargetId(domain.getRoid())
+        .setClientId(domain.getClID())
+        .setEventTime(domain.getExDate())
+        .setRecurrenceEndTime(END_OF_TIME)
+        .setParent(historyEntry)
+        .build();
+    ofy().transact(new VoidWork() {
+
+      @Override
+      public void vrun() {
+        ofy().save().entity(billingEvent);
+      }
+    });
+    return billingEvent;
+  }
+
+  private static PollMessage createAutoRenewPollMessage(
+      XjcRdeDomain domain, HistoryEntry historyEntry) {
+    final PollMessage pollMessage = new PollMessage.Autorenew.Builder()
+        .setTargetId(domain.getRoid())
+        .setClientId(domain.getClID())
+        .setEventTime(domain.getExDate())
+        .setMsg("Domain was auto-renewed.")
+        .setParent(historyEntry)
+        .build();
+    ofy().transact(new VoidWork() {
+
+      @Override
+      public void vrun() {
+        ofy().save().entity(pollMessage);
+      }
+    });
+    return pollMessage;
+  }
+
+  private static HistoryEntry createHistoryEntry(XjcRdeDomain domain) {
+    // TODO: (wolfgang) validate the missing fields in code review
+    XjcRdeDomainElement element = new XjcRdeDomainElement(domain);
+    final HistoryEntry historyEntry = new HistoryEntry.Builder()
+        .setType(HistoryEntry.Type.RDE_IMPORT)
+//        .setPeriod(getCommandPeriod())
```

Comment:
These are properties that I'm not sure need to be included. I think they are valid for epp messages but not for rde import. Is it ok to omit them? If not, any suggestions on what values should be added here?